packages/terminal-external/src/electron-node/terminal-external-service.ts

```diff
@@ -0,0 +1,195 @@
+        const terminalConfig = configuration['terminal.external.osxExec'];
+        const terminalApp = terminalConfig || MacTerminalExternalService.getDefaultTerminalOSX();
+        return new Promise<void>((c, e) => {
+            const args = ['-a', terminalApp];
+            if (cwd) {
+                args.push(cwd);
+            }
+            const child = cp.spawn(MacTerminalExternalService.osxOpener, args);
+            child.on('error', e);
+            child.on('exit', () => c());
+        });
+    }
+
+    async getDefaultExec(): Promise<string> {
+        return MacTerminalExternalService.getDefaultTerminalOSX();
+    }
+
+    /**
+     * Get the default terminal app on OSX.
+     */
+    public static getDefaultTerminalOSX(): string {
+        return this.defaultTerminalApp;
+    }
+}
+
+@injectable()
+export class LinuxTerminalExternalService implements TerminalExternalService {
+    private static DEFAULT_TERMINAL_LINUX_READY: Promise<string>;
+
+    async openTerminal(configuration: TerminalExternalConfiguration, cwd: string): Promise<void> {
+        await this.spawnTerminal(configuration, FileUri.fsPath(cwd));
+    }
+
+    private async spawnTerminal(configuration: TerminalExternalConfiguration, cwd?: string): Promise<void> {
+        const terminalConfig = configuration['terminal.external.linuxExec'];
+        const execPromise = terminalConfig ? Promise.resolve(terminalConfig) : LinuxTerminalExternalService.getDefaultTerminalLinux();
+
+        return new Promise<void>((c, e) => {
+            execPromise.then(exec => {
+                const env = cwd ? { cwd } : undefined;
+                const child = cp.spawn(exec, [], env);
+                child.on('error', e);
+                child.on('exit', () => c());
+            });
+        });
+    }
+
+    async getDefaultExec(): Promise<string> {
+        return LinuxTerminalExternalService.getDefaultTerminalLinux();
+    }
+
+    /**
+     * Get the default terminal app on Linux.
+     * (determine and initialize the variable based on desktop environment if not found)
+     */
+    public static async getDefaultTerminalLinux(): Promise<string> {
+        if (!LinuxTerminalExternalService.DEFAULT_TERMINAL_LINUX_READY) {
+            LinuxTerminalExternalService.DEFAULT_TERMINAL_LINUX_READY = new Promise(async r => {
+                if (OS.type() === OS.Type.Linux) {
+                    const isDebian = await fs.pathExists('/etc/debian_version');
+                    if (isDebian) {
+                        r('x-terminal-emulator');
+                    } else if (process.env.DESKTOP_SESSION === 'gnome' || process.env.DESKTOP_SESSION === 'gnome-classic') {
+                        r('gnome-terminal');
+                    } else if (process.env.DESKTOP_SESSION === 'kde-plasma') {
+                        r('konsole');
+                    } else if (process.env.COLORTERM) {
+                        r(process.env.COLORTERM);
```

Comment:
This seems to fail for me (on RedHat Linux) at this branch. It seems that the `COLORTERM` in my environment is `truecolor`, but when that's used to spawn a terminal, the `openTerminal` request fails. 

```
root ERROR Request openTerminal failed with error: spawn truecolor ENOENT Error: spawn truecolor ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:264:19)
    at onErrorNT (internal/child_process.js:456:16)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
```

VSCode suffers the same failure, though, so perhaps we shouldn't expect to do better, and once I set the preference to the correct path, I do get a new terminal as expected.

Reply:
https://www.gnu.org/software/gettext/manual/html_node/The-TERM-variable.html

Not sure what the actual semantic of `TERM` and `COLORTERM` is, but IIUC those are flags one can provide to programs like `xterm` to configure some kind of feature support, or in the case of `COLORTERM` it would be set by the terminal application itself for processes spawned by it to detect features like `truecolor` support.

Being as bad as VS Code is acceptable, but if someone is able to tell for sure that this is not the right way to do it then we can also fix that logic.